### PR TITLE
compute id and is together

### PR DESCRIPTION
**Pull request by mlgarchery**

### CODE REVIEW

**Comment by mlgarchery** [comment]
Benchmarks seem to be ok. I suggest the config to be returned along with the scene (in stored.go). This way the stored assets are truly independant and we will be able to use them in the benchmark. I am opening an issue

File: render/cast.go [hunk 1]
```diff
@@ -8,69 +8,75 @@ import (
...
+			if ps_specular > 0 {
+				is += ims * math.Pow(ps_specular, a)
+			}
 		}
 	}
-	return i
+
+	if id < 0 {
+		is = 0
+	}
```
Comment (QGarchery) [inline render/cast.go:69]: It should be checked and set to 0 for each light: if a light is not in the right direction then its contribution to `is` is 0

**Review by QGarchery — COMMENTED** [review]
Maybe we can actually keep the computation for `id` and `is` separated, but the computation for `is` reuses the intersection found by the computation for `id`

File: render/cast.go (hunk 1 above)
Comment (mlgarchery) [reply render/cast.go]: I added the condition `ps_diffuse >= 0` before incrementing `is`

**Comment by mlgarchery** [comment]
> Maybe we can actually keep the computation for `id` and `is` separated, but the computation for `is` reuses the intersection found by the computation for `id`

I think we can keep the intensity computation in one function, `is` and `id` are somewhat coupled and use some common variables. Also, it keeps us from iterating over the lights twice.

**Pull request merged**